components/Tabs/Tabs.tsx

```diff
@@ -105,48 +119,54 @@ export const Tabs = ({
    */
   const uniquePrefix = `${id || tabGroupId}-`;
 
+  const tabItemsWithSlug = tabItems.map((tabItem) => ({
+    ...tabItem,
+    slug: slugify(tabItem.label),
+  }));
+
   const handleTabChange = (index: number) => {
     setSelectedIndex(index);
-    const tabHash = `#${uniquePrefix}${slugify(tabItems[index].label)}`;
+    const tabHash = `#${uniquePrefix}${tabItemsWithSlug[index].slug}`;
     window.history.replaceState(null, '', tabHash); // Update hash without adding to history
   };
 
   // Check URL hash on initial load, update the selected tab and scroll to the correct position
   useEffect(() => {
-    const pageHash = window.location.hash.slice(1); // Remove the "#" from the hash
+    // Remove the "#" from the hash
+    const pageHash = window.location.hash.slice(1);
 
     // Check if the current page hash starts with the unique prefix
     if (pageHash.startsWith(uniquePrefix)) {
-      // Remove the unique prefix from the page hash
       const strippedHash = pageHash.replace(uniquePrefix, '');
-
-      // Find the index of the tab item with a tab hash that matches the stripped page shash
-      const index = tabItems.findIndex(tabItem => slugify(tabItem.label) === strippedHash);
+      // Find the index of the tab item with a tab slug that matches the stripped page hash
+      const index = tabItemsWithSlug.findIndex(tabItem => tabItem.slug === strippedHash);
 
       if (index !== -1) {
-        /**
-         * For SM breakpoint and above, if the page hash matches a tab hash,
-         * set that tab as active and scroll to the top of the correct tab group
-         */
-        if (isRenderTabs) {
-          setSelectedIndex(index);
-          tabGroupRef.current?.scrollIntoView({ behavior: 'smooth', block: 'start' });
-        }
-        // On mobile (XS), scroll to the id anchor at the top of the exposed item content
-        else {
-          const element = document.getElementById(`#${uniquePrefix}${slugify(tabItems[index].label)}`);
-          element?.scrollIntoView({ behavior: 'smooth', block: 'start' });
-        }
+        setSelectedIndex(index);
+
+        setTimeout(() => {
+          /**
+           * For SM breakpoint and above, if the page hash matches a tab hash,
+           * set that tab as active and scroll to the top of the correct tab group
+           */
+          if (isRenderTabs) {
+            tabGroupRef.current?.scrollIntoView({ behavior: 'smooth', block: 'start' });
+          } else {
+            const element = document.getElementById(`${uniquePrefix}${tabItemsWithSlug[index].slug}`);
+            element?.scrollIntoView({ behavior: 'smooth', block: 'start' });
+          }
+        }, 0); // Ensures the DOM is fully loaded before scrolling
```

Comment:
Can you tell me more about this setTimeout call? This is an old domReady hack if I recall.

Reply:
Actually I guess I don't really need this. I'll remove this setTimeout in my next PR 👍🏼